src/main/java/com/moujounakki/archimedesfluids/mixins/MixinFlowingFluid.java

```diff
@@ -1,87 +1,138 @@
-                BlockPos pos1 = pos.relative(direction);
-                BlockState blockstate1 = level.getBlockState(pos1);
-                FluidSpreadType spreadType1 = this.getFluidSpreadType(blockstate1);
-                
-                if (spreadType1 == FluidSpreadType.REPLACE) {
-                    if (!blockstate1.isAir()) {
-                        this.beforeDestroyingBlock(level, pos1, blockstate1);
-                    }
-                    this.transferFluid(level, pos, pos1);
-                    break;
+        }
+    } else if (amount == 1 && level.random.nextFloat() < 0.3) {
+        List<Direction> directionList = Direction.Plane.HORIZONTAL.shuffledCopy(level.getRandom());
+        Direction[] shuffledDirections = directionList.toArray(new Direction[0]);
+        for (Direction direction : shuffledDirections) {
+            BlockPos pos1 = pos.relative(direction);
+            BlockState blockstate1 = level.getBlockState(pos1);
+            FluidSpreadType spreadType1 = this.getFluidSpreadType(blockstate1);
+
+            if (spreadType1 == FluidSpreadType.REPLACE) {
+                if (!blockstate1.isAir()) {
+                    this.beforeDestroyingBlock(level, pos1, blockstate1);
                 }
+                this.transferFluid(level, pos, pos1);
+                break;
             }
         }
     }
-    
+
+    // Get the configuration
+    int maxUpdatesPerTick = ArchimedesFluidsCommonConfig.getMaxUpdatesPerTick();
+    int queueCleanInterval = ArchimedesFluidsCommonConfig.getUpdateQueueCleanInterval();
+
+    // Reset the updates processed counter for each tick
+    int updatesProcessed = 0;
+
+    // Process updates from the queue, up to the specified limit.
+    while (updatesProcessed < maxUpdatesPerTick && !updateQueue.isEmpty()) {
+        BlockPos updatePos = updateQueue.poll();
+        // Implement your custom logic here for processing updates from the queue
+        updatesProcessed++;
+    }
+
+    // Enqueue the current position for future processing only if the queue size is below the configured limit
+    if (updateQueue.size() < ArchimedesFluidsCommonConfig.getMaxUpdateQueueSize()) {
+        updateQueue.offer(pos);
+    } else {
+        // Optionally, you can log a message or take some action when the queue is full
+        // For example, you can skip adding to the queue or remove old entries
+    }
+
+    // Optionally clean the queue at regular intervals
+    if (queueCleanInterval > 0 && level.getGameTime() % queueCleanInterval == 0) {
+        cleanQueue();  // Implement this method based on your queue cleaning logic
+    }
+}
+
+private void cleanQueue() {
+    final int MAX_QUEUE_SIZE = ArchimedesFluidsCommonConfig.getMaxUpdateQueueSize(); // Get the max queue size from the config
+
+    // Limit the size of the queue
+    while (updateQueue.size() > MAX_QUEUE_SIZE) {
+        updateQueue.poll(); // Remove the oldest elements to maintain the size limit
+    }
```

Comment:
Is this necessary since the queue can never grow over the max size?

Reply:
That was more of a just in case section, and to keep logic from being lost to really old requests. It made more sense to reallocate the oldest requests to make way for newer ones so that the flowing logic didn't have massive amounts of input lag before an interaction (like placing a bucket of water or block) and that it would be less noticable in the offchance the queue max was hit. For instance, if that was not there, and the max queue was hit when placing a water bucket, that update would just be discarded and lead to unintended and frankly, clearly noticable issue by the end user. With that check, the water bucket gets placed as normal and meshes into the other updates which were old enough to probably have been rendered unnecssary anyway.